Assets/HoloToolkit/UX/Scripts/Buttons/CompoundButtonIcon.cs

```diff
@@ -0,0 +1,313 @@
+                c.a = alpha;
+                instantiatedMaterial.SetColor(alphaColorProperty, c);
+            }
+        }
+
+        private void ClearInstancedAssets()
+        {
+            // Prevent material leaks
+            if (instantiatedMaterial != null)
+            {
+                if (Application.isPlaying)
+                    GameObject.Destroy(instantiatedMaterial);
+                else
+                    GameObject.DestroyImmediate(instantiatedMaterial);
+
+                instantiatedMaterial = null;
+            }
+            if (instantiatedMesh != null)
+            {
+                if (Application.isPlaying)
+                    GameObject.Destroy(instantiatedMesh);
+                else
+                    GameObject.DestroyImmediate(instantiatedMesh);
+
+                instantiatedMesh = null;
+            }
+
+            // Reset to default mats and meshes
+            if (Profile != null)
+            {
+                if (IconRenderer != null)
+                {
+                    // Restore the icon material to the renderer
+                    IconRenderer.sharedMaterial = Profile.IconMaterial;
+                }
+                if (IconMeshFilter != null)
+                {
+                    IconMeshFilter.sharedMesh = Profile.IconMesh;
+                }
+            }
+            // Reset our alpha to the alpha target
+            alpha = alphaTarget;
+        }
+
+        private IEnumerator UpdateAlpha()
+        {
+            float startTime = Time.time;
+            Color color = Color.white;
+            string alphaColorProperty = string.IsNullOrEmpty(Profile.AlphaColorProperty) ? "_Color" : Profile.AlphaColorProperty;
+            if (instantiatedMaterial != null)
+            {
+                color = instantiatedMaterial.GetColor(alphaColorProperty);
+                color.a = alpha;
+            }
+            while (Time.time < startTime + Profile.AlphaTransitionSpeed)
+            {
+                alpha = Mathf.Lerp(alpha, alphaTarget, (Time.time - startTime) / Profile.AlphaTransitionSpeed);
+                if (instantiatedMaterial != null && !string.IsNullOrEmpty(alphaColorProperty))
+                {
+                    color.a = alpha;
+                    instantiatedMaterial.SetColor(alphaColorProperty, color);
+                }
+                yield return null;
+            }
+            alpha = alphaTarget;
+            RefreshAlpha();
+            updatingAlpha = false;
+            yield break;
+        }
+
+        [SerializeField]
+        private Texture2D iconOverride;
+
+        [SerializeField]
+        private string iconName;
+
+        [SerializeField]
+        private float alpha = 1f;
+
+        [SerializeField]
+        private MeshRenderer targetIconRenderer;
+
+        private Material instantiatedMaterial;
+        private Mesh instantiatedMesh;
+        private bool updatingAlpha = false;
+        private float alphaTarget = 1f;
```

Comment:
Why are all these fields at the bottom of the class?

Reply:
Yeah, moved to top.  I think Lars likes putting things on the bottom :).